### PR TITLE
Add local staging maven jars publishing and add custom build.sh

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add local staging maven jars publishing and add custom build.sh

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/3743
https://github.com/opensearch-project/opensearch-build/issues/3971

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
```

[~/opensearch-build-peterzhuamazon/tar/builds/opensearch/maven/org/opensearch]
% ls plugin/custom-codecs/2.10.0.0
custom-codecs-2.10.0.0.pom      custom-codecs-2.10.0.0.pom.sha1    custom-codecs-2.10.0.0.pom.sha512  custom-codecs-2.10.0.0.zip.md5   custom-codecs-2.10.0.0.zip.sha256
custom-codecs-2.10.0.0.pom.md5  custom-codecs-2.10.0.0.pom.sha256  custom-codecs-2.10.0.0.zip         custom-codecs-2.10.0.0.zip.sha1  custom-codecs-2.10.0.0.zip.sha512

[~/opensearch-build-peterzhuamazon/tar/builds/opensearch/maven/org/opensearch]
% ls opensearch-custom-codecs/2.10.0.0
opensearch-custom-codecs-2.10.0.0.jar         opensearch-custom-codecs-2.10.0.0-javadoc.jar         opensearch-custom-codecs-2.10.0.0.module         opensearch-custom-codecs-2.10.0.0.pom         opensearch-custom-codecs-2.10.0.0-sources.jar
opensearch-custom-codecs-2.10.0.0.jar.md5     opensearch-custom-codecs-2.10.0.0-javadoc.jar.md5     opensearch-custom-codecs-2.10.0.0.module.md5     opensearch-custom-codecs-2.10.0.0.pom.md5     opensearch-custom-codecs-2.10.0.0-sources.jar.md5
opensearch-custom-codecs-2.10.0.0.jar.sha1    opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha1    opensearch-custom-codecs-2.10.0.0.module.sha1    opensearch-custom-codecs-2.10.0.0.pom.sha1    opensearch-custom-codecs-2.10.0.0-sources.jar.sha1
opensearch-custom-codecs-2.10.0.0.jar.sha256  opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha256  opensearch-custom-codecs-2.10.0.0.module.sha256  opensearch-custom-codecs-2.10.0.0.pom.sha256  opensearch-custom-codecs-2.10.0.0-sources.jar.sha256
opensearch-custom-codecs-2.10.0.0.jar.sha512  opensearch-custom-codecs-2.10.0.0-javadoc.jar.sha512  opensearch-custom-codecs-2.10.0.0.module.sha512  opensearch-custom-codecs-2.10.0.0.pom.sha512  opensearch-custom-codecs-2.10.0.0-sources.jar.sha512

```

**Comment by reta** [comment]
@peterzhuamazon thanks but why we need Jenkins since we have GH workflow for it (that all other repos use)? We will need to copy/paste the script all over the place :(

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon thanks but why we need Jenkins since we have GH workflow for it (that all other repos use)? We will need to copy/paste the script all over the place :(

This is release maven even though the name says staging local.
It will be copied to S3 and will be published to maven central when 2.10.0 release.
Sorry for the confusing naming tho.

Also waiting for this from you to get merged beforehand:
* https://github.com/opensearch-project/custom-codecs/pull/26

**Comment by reta** [comment]
> @peterzhuamazon thanks but why we need Jenkins since we have GH workflow for it (that all other repos use)? We will need to copy/paste the script all over the place :(

Got the answer, this is the release workflows

**Comment by reta** [comment]
Please hold on till https://github.com/opensearch-project/custom-codecs/pull/26 gets in to have the right articfat names

File: build.gradle [hunk 1]
```diff
@@ -43,7 +42,7 @@ repositories {
 }
 
 allprojects {
-  group 'org.opensearch.plugin'
+  group 'org.opensearch'
```
Comment (gaiksaya) [inline build.gradle:45]: Should be under plugin.

Comment (peterzhuamazon) [reply build.gradle:45]: no, it should be non-plugin for all project, and plugin for only the plugin zip.

Comment (gaiksaya) [reply build.gradle:45]: I see what you mean https://github.com/opensearch-project/custom-codecs/pull/27/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R77

**Review by gaiksaya — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (reta) [reply build.gradle:45]: Wait a sec, for plugins we publish `-client` artifacts and they are all under `org.opensearch.plugin`, for example:

https://mvnrepository.com/artifact/org.opensearch.plugin/ingest-attachment/2.9.0

Comment (reta) [reply build.gradle:45]: For info, looked with @peterzhuamazon : for non-core plugins, JAR artifacts go under `org.opensearch` but ZIP artifacts go under  `org.opensearch.plugin`. For core plugins, everything goes under ` org.opensearch.plugin`

**Review by reta — APPROVED** [review]

**Pull request merged**